### PR TITLE
Add IAM client

### PR DESCRIPTION
**Pull request by jderusse**
For now just CRUD operation on `user`

https://twitter.com/benjamindavies/status/1258300350018322435

### CODE REVIEW

File: src/Core/src/AwsClientFactory.php [hunk 1]
```diff
@@ -152,6 +153,19 @@ public function eventBridge(): EventBridgeClient
         return $this->serviceCache[__METHOD__];
     }
 
+    public function iam(): DynamoDbClient
```
Comment (Nyholm) [inline src/Core/src/AwsClientFactory.php:156]:
```suggestion
    public function iam(): IamClient
```

File: src/Service/Iam/README.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+## Install
+
+```cli
+composer require async-aws/iam
+```
+
+## Documentation
+
+See https://async-aws.com/clients/iam.html for documentation.
```
Comment (Nyholm) [inline src/Service/Iam/README.md:16]:
```suggestion
See https://async-aws.com for documentation.
```

Comment (jderusse) [reply src/Service/Iam/README.md]: thanks. Fixed template too

File: src/Core/src/AwsClientFactory.php (hunk 1 above)
Comment (jderusse) [reply src/Core/src/AwsClientFactory.php]: thanks

File: src/Service/Iam/README.md (hunk 2 above)
Comment (jderusse) [reply src/Service/Iam/README.md]: hmm.. no in fact, the issue is not here, the issue is the documentation is missing.

Reverted.

Comment (Nyholm) [reply src/Service/Iam/README.md]: Yeah, but documentation is not strictly required. Is it?

**Comment by jderusse** [comment]
Beware, this PR also contain a fix on code generator to handle global endpoint (no region in endpoint)
It should be BC compatible

File: src/Service/Iam/README.md (hunk 2 above)
Comment (jderusse) [reply src/Service/Iam/README.md:16]: I think it should. At least for SEO, and some people (me included) use the list of ApiClient menu to known the liste of implemented client.

Comment (Nyholm) [reply src/Service/Iam/README.md:16]: It is currently required to be in the list, but it is not required to have a separate page for it.

File: docs/clients/index.md [hunk 3]
```diff
@@ -140,7 +140,8 @@ for more information.
 | CodeDeploy                                  | [async-aws/code-deploy](https://packagist.org/packages/async-aws/code-deploy)                             |
 | [CognitoIdentityProvider](./cognito-idp.md) | [async-aws/cognito-identity-provider](https://packagist.org/packages/async-aws/cognito-identity-provider) |
 | [DynamoDb]('./dynamodb.md)                  | [async-aws/dynamo-db](https://packagist.org/packages/async-aws/dynamo-db)                                 |
-| EventBridge                                 | [async-aws/lambda](https://packagist.org/packages/async-aws/event-bridge)                                       |
+| EventBridge                                 | [async-aws/lambda](https://packagist.org/packages/async-aws/event-bridge)                                 |
```
Comment (Nyholm) [inline docs/clients/index.md:143]:
```suggestion
| EventBridge                                 | [async-aws/event-bridge](https://packagist.org/packages/async-aws/event-bridge)                                 |
```

Comment (jderusse) [reply docs/clients/index.md]: fixed

File: phpstan-baseline.neon [hunk 4]
```diff
@@ -1,7 +1,2 @@
 parameters:
-	ignoreErrors:
-		-
-			message: "#^Parameter \\#1 \\$input of class AsyncAws\\\\Ses\\\\ValueObject\\\\EmailContent constructor expects array\\(\\)\\|array\\('Simple' \\=\\> array\\|AsyncAws\\\\Ses\\\\ValueObject\\\\Message\\|null, 'Raw' \\=\\> array\\|AsyncAws\\\\Ses\\\\ValueObject\\\\RawMessage\\|null, 'Template' \\=\\> array\\|AsyncAws\\\\Ses\\\\ValueObject\\\\Template\\|null\\), array\\('Raw' \\=\\> AsyncAws\\\\Ses\\\\ValueObject\\\\RawMessage\\) given\\.$#"
-			count: 1
-			path: src/Integration/Laravel/Mail/src/Transport/AsyncAwsSesTransport.php
-
+	ignoreErrors: []
```
Comment (jderusse) [inline phpstan-baseline.neon:2]: shouldn't we remove this file?

Comment (Nyholm) [reply phpstan-baseline.neon:2]: Sure

Comment (jderusse) [reply phpstan-baseline.neon]: done

**Comment by jderusse** [comment]
thanks for the review

**Pull request merged**